### ISSUE TITLE
Database use optimisations

### ISSUE THREAD

**Issue by adekbadek** [Automattic/newspack-popups#478]
Currently, the Lightweight API is not so lightweight on the DB. Some ideas for improvements:

- [x] **Empty client data**. For a first-time visitor to a non-post page, client data entry is created. This is not needed, unless they are a donor or a subscriber (#490). 
- [x] Currently, a row is created in the options table per client-prompt pair (e.g. with the information about prompts suppression). It'd be better to create save that information in the row which is created for the client (e.g. with information about posts read). In essence, it's about creating **not more than one row per client** (#490).
- The rows in the options table are never **purged**. The data in the custom table (for segmentation purposes) is removed after 30 days - but this is just data about read posts, which can be discarded. The subscriber & donor status should be retained. 
    - [ ] Perhaps a good solution would be to save the last client update in the client data, and remove rows (client data) not updated in more than ~two months (enabled by #508, but deploying that change would be possible only after changes from #508 are live for enough time). 
    - [x] The preview session data can safely be purged anytime (#508)
- [x] Client data is saved for **bot traffic** #509

cc @dkoo @jeffersonrabb

**Comment by dkoo** [comment]
Also based on a recent high-traffic event:

[ ] Write client data to a custom table instead of the WP options table

**Comment by adekbadek** [comment]
> We're using wp_cache_set and wp_cache_get and inserting their values directly into the options table. Why aren't we using the actual Transients API?

The direct use of the DB stems from the API being separate from WP for performance reasons.

**Comment by adekbadek** [comment]
Putting this on hold, as we can only proceed with the stale client data removal after the update-tracking changes (#508) are live for about two months.

**Comment by adekbadek** [comment]
> The rows in the options table are never purged. 

Better purging added in #584
